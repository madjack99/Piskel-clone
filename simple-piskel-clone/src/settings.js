const settings = {
  primaryColor: '#000000',
  secondaryColor: '#ff000',
  realCanvasResolution: 512,
  canvasSelectedResolution: 32,
  brushSize: 1,
  drawingTool: 'pencil',
};

export default settings;
