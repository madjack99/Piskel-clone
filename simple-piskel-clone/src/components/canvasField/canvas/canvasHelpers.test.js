import testDom from '../../../test/testDom';
import settings from '../../../settings';

describe('rgb to hex function', () => {
  document.body.innerHTML = testDom;
  const { rgbToHex } = require('./canvasHelpers');

  test('should return correct color value', () => {
    expect(rgbToHex(44, 22, 88)).toBe('#2c1658');
  });

  test('should return #0 if called without args', () => {
    expect(rgbToHex()).toBe('#0');
  });
});

test('canvas should be painted with white after applying fillWithWhite func', () => {
  document.body.innerHTML = testDom;

  const { fillCanvasWithWhite } = require('./canvasHelpers');
  const canvas = document.querySelector('.canvas');

  fillCanvasWithWhite(canvas);

  const ctx = canvas.getContext('2d');
  // expect(ctx.__getEvents()[0].props.value).toBe('#fff');
  expect(ctx.__getEvents()).toMatchSnapshot();
});

test('should correctly update the framesImagesArr', () => {
  document.body.innerHTML = testDom;

  const { saveImageFromMainCanvas } = require('./canvasHelpers');
  const { framesImagesArr } = settings;

  saveImageFromMainCanvas(0);

  expect(framesImagesArr.length).toBe(1);
});
