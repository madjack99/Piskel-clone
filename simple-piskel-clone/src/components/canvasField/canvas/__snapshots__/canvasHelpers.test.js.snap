// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`canvas should be painted with white after applying fillWithWhite func 1`] = `
Array [
  Object {
    "props": Object {
      "value": "#fff",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillStyle",
  },
  Object {
    "props": Object {
      "height": 128,
      "width": 128,
      "x": 0,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
]
`;
